Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -100,25 +172,52 @@ public void Generate_LicenseAcceptanceRequiredByMainPackage()
+            // 1. a) Only target package. User does not accept -> fails with error
+            ProcessedArgs args = CreateArgs("parent.requiredAccept.id", "1.0", "cs", null, false /* accept licenses */ , 
+                false, outputDir);
             bool result = apg.Generate(args);
             Assert.IsFalse(result, "Generator should fail because license has not been accepted");
 
             logger.AssertSingleErrorExists("parent.requiredAccept.id", "1.0"); // error listing the main package
             logger.AssertSingleWarningExists("parent.requiredAccept.id", "1.0"); // warning for each licensed package
             logger.AssertWarningsLogged(1);
 
-            // 2. User accepts -> succeeds with warnings
+            // 1. b) Target package and dependencies. User does not accept -> fails with error
+            logger.Reset();
+            args = CreateArgs("parent.requiredAccept.id", "1.0", "cs", null, false /* accept licenses */ ,
+                true /* generate plugins for dependencies */, outputDir);
+            result = apg.Generate(args);
+            Assert.IsFalse(result, "Generator should fail because license has not been accepted");
+
+            logger.AssertSingleErrorExists("parent.requiredAccept.id", "1.0"); // error listing the main package
+            logger.AssertSingleWarningExists("parent.requiredAccept.id", "1.0"); // warning for each licensed package
+            logger.AssertWarningsLogged(1);
+
+            // 2. a) Only target package. User accepts -> succeeds with warnings
             logger.Reset();
-            args = CreateArgs("parent.requiredAccept.id", "1.0", "cs", null, true /* accept licenses */ , outputDir);
+            args = CreateArgs("parent.requiredAccept.id", "1.0", "cs", null, true /* accept licenses */ , 
+                false, outputDir);
             result = apg.Generate(args);
             Assert.IsTrue(result, "Generator should succeed if licenses are accepted");
 
             logger.AssertSingleWarningExists(UIResources.APG_NGAcceptedPackageLicenses); // warning that licenses accepted
             logger.AssertSingleWarningExists("parent.requiredAccept.id", "1.0"); // warning for each licensed package
             logger.AssertWarningsLogged(2);
             logger.AssertErrorsLogged(0);
+
+            // 2. b) Target package and dependencies. User accepts -> succeeds with warnings
+            clearDirectory(outputDir);
+            logger.Reset();
+            args = CreateArgs("parent.requiredAccept.id", "1.0", "cs", null, true /* accept licenses */ ,
+                true /* generate plugins for dependencies */, outputDir);
+            result = apg.Generate(args);
+            Assert.IsTrue(result, "Generator should succeed if licenses are accepted");
+
+            logger.AssertSingleWarningExists(UIResources.APG_RecurseEnabled_SQALENotEnabled); // warning that licenses accepted
```

Comment:
The comment is not accurate, and check isn't really connected to the purpose of this test. Aren't there any tests covering the SQALE generation?